vir/src/context.rs

```diff
@@ -90,23 +98,42 @@ impl<'tcx> VirCtxt<'tcx> {
     }
 
     pub fn mk_true(&'tcx self) -> Expr<'tcx> {
-        self.alloc(ExprData::Const(self.alloc(ConstData::Bool(true))))
+        self.mk_const_expr(self.alloc(ConstData::Bool(true)))
```

Comment:
Unrelated to this PR, but we should remember: these kind of nodes (`true`, `false`, `0`, `1`, …) should only need to be allocated once and re-used throughout.

Reply:
They shouldn't need to be allocated at all actually: they can just be `&'static`. However, with the current `Lazy` variant having `Fn(&'vir VirCtxt...` it makes `'vir` invariant and so this isn't possible. Imo we should get rid of `Lazy` (it can be replaced with a `with_let_bound_vars(|vars| ...)` where we never even need to `reify`, but let's discuss that.